service-front/module/Application/src/ControllerFactory/ControllerAbstractFactory.php

```diff
@@ -133,18 +133,22 @@ public function __invoke(ContainerInterface $container, $requestedName, array $o
         $controllerName = $this->getControllerName($requestedName);
 
         $formElementManager = $container->get('FormElementManager');
+        // Var not used within class, but required to populate twig helper function within templates.
+        $sessionDetails = $container->get('PersistentSessionDetails');
```

Comment:
I'm not sure about this comment: how is this used in the twig helper? Is it acting somehow like a global variable?

Reply:
Unless the container I've constructed is called, there won't be anything contained within it. I've had to place this low enough so it's getting the right routing information, but high enough that it doesn't interfere with anything. My comment doesn't explain it very well. It's a little hacky, to be honest.

Reply:
Right, so how is `$sessionDetails` called by the template? Are you effectively just initialising the PersistentSessionDetails (in which case the variable is not necessary)?

Reply:
Oh, it's just initiated. I put it in a var out of habit more than anything, I'll change that.

Also, in reply to your other comment, as I can't reply to that directly. The acordin is loaded, I think, on every page. But only takes effect in a few circumstances. Mainly when it's used in either creation of the LPA, so it can display additional guidance or when they first start the LPA journey to skip the guidance at the top of the page regarding LPA types.